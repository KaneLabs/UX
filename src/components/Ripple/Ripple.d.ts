export default class Ripple extends React.PureComponent<any, any, any> {
    static defaultProps: any;
    static propTypes: any;
    constructor(props: any);
    onLayout(event: any): void;
    onPress(event: any): void;
    onPressIn(event: any): void;
    onPressOut(event: any): void;
    onLongPress(event: any): void;
    onAnimationEnd(event: any): void;
    renderRipple({ unique, progress, locationX, locationY, R, }: {
        unique: any;
        progress: any;
        locationX: any;
        locationY: any;
        R: any;
    }): JSX.Element;
    unique: number;
    mounted: boolean;
    rippleFades: any;
    isPressingIn: boolean;
    animationWaitingForEnd: boolean;
    webGetPositionInElement: (e: any) => {
        x: number;
        y: number;
    };
    signalAnimationEnd(): void;
    forceAnimationEnd(): void;
    startRipple(event: any): void;
}
import React from "react";
