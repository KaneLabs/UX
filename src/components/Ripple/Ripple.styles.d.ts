export const radius: 10;
export default styles;
declare namespace styles {
    const container: {
        backgroundColor: string;
        overflow: string;
        zIndex: number;
        position: "absolute";
        left: 0;
        right: 0;
        top: 0;
        bottom: 0;
    };
    namespace ripple {
        export const width: number;
        export const height: number;
        export { radius as borderRadius };
        export const overflow: string;
        export const position: string;
        export const zIndex: number;
    }
}
