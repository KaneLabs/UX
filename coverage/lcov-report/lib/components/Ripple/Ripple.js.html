
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for lib/components/Ripple/Ripple.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">lib/components/Ripple</a> Ripple.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">44.57% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>82/184</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">18% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>9/50</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">28% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>7/25</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">2x</span></td><td class="text"><pre class="prettyprint lang-js">var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _extends2=_interopRequireDefault(require("@babel/runtime/helpers/extends"));var _objectWithoutProperties2=_interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));var _classCallCheck2=_interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));var _createClass2=_interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _assertThisInitialized2=_interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));var _inherits2=_interopRequireDefault(require("@babel/runtime/helpers/inherits"));var _possibleConstructorReturn2=_interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));var _getPrototypeOf2=_interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));var _propTypes=_interopRequireDefault(require("prop-types"));var _react=_interopRequireWildcard(require("react"));var _View=_interopRequireDefault(require("react-native-web/dist/exports/View"));var _Animated=_interopRequireDefault(require("react-native-web/dist/exports/Animated"));var _Easing=_interopRequireDefault(require("react-native-web/dist/exports/Easing"));var _TouchableWithoutFeedback=_interopRequireDefault(require("react-native-web/dist/exports/TouchableWithoutFeedback"));var _Platform=_interopRequireDefault(require("react-native-web/dist/exports/Platform"));var _Ripple=_interopRequireWildcard(require("./Ripple.styles"));var _jsxFileName="/Users/kane/Lab/Eros/src/components/Ripple/Ripple.js";function _createSuper(Derived){var hasNativeReflectConstruct=_isNativeReflectConstruct();return function _createSuperInternal(){var Super=(0,_getPrototypeOf2.default)(Derived),result;<span class="missing-if-branch" title="else path not taken" >E</span>if(hasNativeReflectConstruct){var NewTarget=(0,_getPrototypeOf2.default)(this).constructor;result=Reflect.construct(Super,arguments,NewTarget);}else{<span class="cstat-no" title="statement not covered" >result=Super.apply(this,arguments);}</span>return(0,_possibleConstructorReturn2.default)(this,result);};}function _isNativeReflectConstruct(){<span class="missing-if-branch" title="if path not taken" >I</span>if(typeof Reflect==="undefined"||!Reflect.construct)<span class="cstat-no" title="statement not covered" >return false;<span class="missing-if-branch" title="if path not taken" >I</span>i</span>f(Reflect.construct.sham)<span class="cstat-no" title="statement not covered" >return false;<span class="missing-if-branch" title="else path not taken" >E</span>i</span>f(typeof Proxy==="function")return true;<span class="cstat-no" title="statement not covered" >try{<span class="cstat-no" title="statement not covered" >Boolean.prototype.valueOf.call(Reflect.construct(Boolean,[],<span class="fstat-no" title="function not covered" >fu</span>nction(){}));<span class="cstat-no" title="statement not covered" >r</span>eturn true;}</span>catch(e){<span class="cstat-no" title="statement not covered" >return false;}</span>}</span>var Ripple=function(_PureComponent){(0,_inherits2.default)(Ripple,_PureComponent);var _super=_createSuper(Ripple);function Ripple(props){var _this;(0,_classCallCheck2.default)(this,Ripple);_this=_super.call(this,props);_this.webGetPositionInElement=<span class="fstat-no" title="function not covered" >fu</span>nction(e){var rect=<span class="cstat-no" title="statement not covered" >e.nativeEvent.target.getBoundingClientRect();</span>var x=<span class="cstat-no" title="statement not covered" >e.nativeEvent.pageX-rect.left;</span>var y=<span class="cstat-no" title="statement not covered" >e.nativeEvent.changedTouches[0].clientY-rect.top;<span class="cstat-no" title="statement not covered" ></span>return{x:x,y:y};}</span>;_this.onLayout=_this.onLayout.bind((0,_assertThisInitialized2.default)(_this));_this.onPress=_this.onPress.bind((0,_assertThisInitialized2.default)(_this));_this.onPressIn=_this.onPressIn.bind((0,_assertThisInitialized2.default)(_this));_this.onPressOut=_this.onPressOut.bind((0,_assertThisInitialized2.default)(_this));_this.onLongPress=_this.onLongPress.bind((0,_assertThisInitialized2.default)(_this));_this.onAnimationEnd=_this.onAnimationEnd.bind((0,_assertThisInitialized2.default)(_this));_this.renderRipple=_this.renderRipple.bind((0,_assertThisInitialized2.default)(_this));_this.unique=0;_this.mounted=false;_this.rippleFades=_this.props.rippleFades&amp;&amp;!_this.props.displayUntilPressOut;_this.isPressingIn=false;_this.animationWaitingForEnd=false;_this.state={width:0,height:0,ripples:[]};return _this;}(0,_createClass2.default)(Ripple,[{key:"componentDidMount",value:function componentDidMount(){this.mounted=true;}},{key:"componentWillUnmount",value:function <span class="fstat-no" title="function not covered" >componentWillUnmount(</span>){<span class="cstat-no" title="statement not covered" >this.mounted=false;}</span>},{key:"onLayout",value:function <span class="fstat-no" title="function not covered" >onLayout(</span>event){var _event$nativeEvent$la=<span class="cstat-no" title="statement not covered" >event.nativeEvent.layout,</span>width=<span class="cstat-no" title="statement not covered" >_event$nativeEvent$la.width,</span>height=<span class="cstat-no" title="statement not covered" >_event$nativeEvent$la.height;</span>var onLayout=<span class="cstat-no" title="statement not covered" >this.props.onLayout;<span class="cstat-no" title="statement not covered" ></span>if(typeof onLayout==='function'){<span class="cstat-no" title="statement not covered" >onLayout(event);}<span class="cstat-no" title="statement not covered" ></span>t</span>his.setState({width:width,height:height});}</span>},{key:"onPress",value:function <span class="fstat-no" title="function not covered" >onPress(</span>event){var onPress=<span class="cstat-no" title="statement not covered" >this.props.onPress;<span class="cstat-no" title="statement not covered" ></span>if(typeof onPress==='function'){<span class="cstat-no" title="statement not covered" >requestAnimationFrame(<span class="fstat-no" title="function not covered" >fu</span>nction(){<span class="cstat-no" title="statement not covered" >return onPress(event);}</span>);}</span>}</span>},{key:"onLongPress",value:function <span class="fstat-no" title="function not covered" >onLongPress(</span>event){var _this$props=<span class="cstat-no" title="statement not covered" >this.props,</span>onLongPress=<span class="cstat-no" title="statement not covered" >_this$props.onLongPress,</span>disabled=<span class="cstat-no" title="statement not covered" >_this$props.disabled;<span class="cstat-no" title="statement not covered" ></span>if(disabled)<span class="cstat-no" title="statement not covered" >return;<span class="cstat-no" title="statement not covered" >i</span></span>f(typeof onLongPress==='function'){<span class="cstat-no" title="statement not covered" >requestAnimationFrame(<span class="fstat-no" title="function not covered" >fu</span>nction(){<span class="cstat-no" title="statement not covered" >return onLongPress(event);}</span>);}<span class="cstat-no" title="statement not covered" ></span>t</span>his.startRipple(event);}</span>},{key:"onPressIn",value:function <span class="fstat-no" title="function not covered" >onPressIn(</span>event){var _this$props2=<span class="cstat-no" title="statement not covered" >this.props,</span>onPressIn=<span class="cstat-no" title="statement not covered" >_this$props2.onPressIn,</span>rippleSequential=<span class="cstat-no" title="statement not covered" >_this$props2.rippleSequential,</span>disabled=<span class="cstat-no" title="statement not covered" >_this$props2.disabled;</span>var ripples=<span class="cstat-no" title="statement not covered" >this.state.ripples;<span class="cstat-no" title="statement not covered" ></span>if(disabled)<span class="cstat-no" title="statement not covered" >return;<span class="cstat-no" title="statement not covered" >t</span></span>his.isPressingIn=true;<span class="cstat-no" title="statement not covered" >i</span>f(!rippleSequential||!ripples.length){<span class="cstat-no" title="statement not covered" >if(onPressIn){<span class="cstat-no" title="statement not covered" >onPressIn(event);}<span class="cstat-no" title="statement not covered" ></span>t</span>his.startRipple(event);}</span>}</span>},{key:"onPressOut",value:function <span class="fstat-no" title="function not covered" >onPressOut(</span>event){var onPressOut=<span class="cstat-no" title="statement not covered" >this.props.onPressOut;<span class="cstat-no" title="statement not covered" ></span>if(typeof onPressOut==='function'){<span class="cstat-no" title="statement not covered" >onPressOut(event);}<span class="cstat-no" title="statement not covered" ></span>t</span>his.signalAnimationEnd();<span class="cstat-no" title="statement not covered" >t</span>his.isPressingIn=false;}</span>},{key:"onAnimationEnd",value:function <span class="fstat-no" title="function not covered" >onAnimationEnd(</span>event){<span class="cstat-no" title="statement not covered" >if(this.props.displayUntilPressOut&amp;&amp;this.isPressingIn){<span class="cstat-no" title="statement not covered" >this.animationWaitingForEnd=true;<span class="cstat-no" title="statement not covered" >r</span>eturn;}<span class="cstat-no" title="statement not covered" ></span>i</span>f(this.props.onAnimationEnd){<span class="cstat-no" title="statement not covered" >this.props.onAnimationEnd(event);}<span class="cstat-no" title="statement not covered" ></span>t</span>his.forceAnimationEnd();}</span>},{key:"signalAnimationEnd",value:function <span class="fstat-no" title="function not covered" >signalAnimationEnd(</span>){<span class="cstat-no" title="statement not covered" >if(this.animationWaitingForEnd){<span class="cstat-no" title="statement not covered" >this.forceAnimationEnd();}</span>}</span>},{key:"forceAnimationEnd",value:function <span class="fstat-no" title="function not covered" >forceAnimationEnd(</span>){<span class="cstat-no" title="statement not covered" >if(this.mounted){<span class="cstat-no" title="statement not covered" >this.setState(<span class="fstat-no" title="function not covered" >fu</span>nction(_ref){var ripples=<span class="cstat-no" title="statement not covered" >_ref.ripples;<span class="cstat-no" title="statement not covered" ></span>return{ripples:ripples.slice(1)};}</span>);}<span class="cstat-no" title="statement not covered" ></span>t</span>his.animationWaitingForEnd=false;}</span>},{key:"startRipple",value:function <span class="fstat-no" title="function not covered" >startRipple(</span>event){var _this$state=<span class="cstat-no" title="statement not covered" >this.state,</span>width=<span class="cstat-no" title="statement not covered" >_this$state.width,</span>height=<span class="cstat-no" title="statement not covered" >_this$state.height;</span>var _this$props3=<span class="cstat-no" title="statement not covered" >this.props,</span>rippleDuration=<span class="cstat-no" title="statement not covered" >_this$props3.rippleDuration,</span>rippleCentered=<span class="cstat-no" title="statement not covered" >_this$props3.rippleCentered,</span>rippleSize=<span class="cstat-no" title="statement not covered" >_this$props3.rippleSize,</span>onRippleAnimation=<span class="cstat-no" title="statement not covered" >_this$props3.onRippleAnimation;</span>var w2=<span class="cstat-no" title="statement not covered" >0.5*width;</span>var h2=<span class="cstat-no" title="statement not covered" >0.5*height;</span>var _event$nativeEvent=<span class="cstat-no" title="statement not covered" >event.nativeEvent,</span>locationX=<span class="cstat-no" title="statement not covered" >_event$nativeEvent.locationX,</span>locationY=<span class="cstat-no" title="statement not covered" >_event$nativeEvent.locationY;<span class="cstat-no" title="statement not covered" ></span>if(_Platform.default.OS==='web'){<span class="cstat-no" title="statement not covered" >locationX=this.webGetPositionInElement(event).x;<span class="cstat-no" title="statement not covered" >l</span>ocationY=this.webGetPositionInElement(event).y;}<span class="cstat-no" title="statement not covered" ></span>i</span>f(rippleCentered){<span class="cstat-no" title="statement not covered" >locationX=w2;<span class="cstat-no" title="statement not covered" >l</span>ocationY=h2;}</span>v</span>ar offsetX=<span class="cstat-no" title="statement not covered" >Math.abs(w2-locationX);</span>var offsetY=<span class="cstat-no" title="statement not covered" >Math.abs(h2-locationY);</span>var R=<span class="cstat-no" title="statement not covered" >rippleSize&gt;0?0.5*rippleSize:Math.sqrt(Math.pow(w2+offsetX,2)+Math.pow(h2+offsetY,2));</span>var ripple=<span class="cstat-no" title="statement not covered" >{unique:this.unique++,progress:new _Animated.default.Value(0),locationX:locationX,locationY:locationY,R:R};</span>var animation=<span class="cstat-no" title="statement not covered" >_Animated.default.timing(ripple.progress,{toValue:1,easing:_Easing.default.out(_Easing.default.ease),duration:rippleDuration,useNativeDriver:true});<span class="cstat-no" title="statement not covered" ></span>onRippleAnimation(animation,this.onAnimationEnd);<span class="cstat-no" title="statement not covered" >t</span>his.setState(<span class="fstat-no" title="function not covered" >fu</span>nction(_ref2){var ripples=<span class="cstat-no" title="statement not covered" >_ref2.ripples;<span class="cstat-no" title="statement not covered" ></span>return{ripples:ripples.concat(ripple)};}</span>);}</span>},{key:"renderRipple",value:function <span class="fstat-no" title="function not covered" >renderRipple(</span>_ref3){var unique=<span class="cstat-no" title="statement not covered" >_ref3.unique,</span>progress=<span class="cstat-no" title="statement not covered" >_ref3.progress,</span>locationX=<span class="cstat-no" title="statement not covered" >_ref3.locationX,</span>locationY=<span class="cstat-no" title="statement not covered" >_ref3.locationY,</span>R=<span class="cstat-no" title="statement not covered" >_ref3.R;</span>var _this$props4=<span class="cstat-no" title="statement not covered" >this.props,</span>rippleColor=<span class="cstat-no" title="statement not covered" >_this$props4.rippleColor,</span>rippleOpacity=<span class="cstat-no" title="statement not covered" >_this$props4.rippleOpacity;</span>var rippleFades=<span class="cstat-no" title="statement not covered" >this.rippleFades;</span>var rippleStyle=<span class="cstat-no" title="statement not covered" >{top:locationY-_Ripple.radius,left:locationX-_Ripple.radius,backgroundColor:rippleColor,transform:[{scale:progress.interpolate({inputRange:[0,1],outputRange:[0.5/_Ripple.radius,R/_Ripple.radius]})}],opacity:rippleFades?progress.interpolate({inputRange:[0,1],outputRange:[rippleOpacity,0]}):rippleOpacity};<span class="cstat-no" title="statement not covered" ></span>return _react.default.createElement(_Animated.default.View,{style:[_Ripple.default.ripple,rippleStyle],key:unique,useNativeDriver:false,__self:this,__source:{fileName:_jsxFileName,lineNumber:244,columnNumber:7}});}</span>},{key:"render",value:function render(){var ripples=this.state.ripples;var onPress=this.onPress,onPressIn=this.onPressIn,onPressOut=this.onPressOut,onLongPress=this.onLongPress,onLayout=this.onLayout;var _this$props5=this.props,delayLongPress=_this$props5.delayLongPress,delayPressIn=_this$props5.delayPressIn,delayPressOut=_this$props5.delayPressOut,disabled=_this$props5.disabled,hitSlop=_this$props5.hitSlop,pressRetentionOffset=_this$props5.pressRetentionOffset,children=_this$props5.children,rippleContainerBorderRadius=_this$props5.rippleContainerBorderRadius,testID=_this$props5.testID,nativeID=_this$props5.nativeID,accessible=_this$props5.accessible,accessibilityLabel=_this$props5.accessibilityLabel,panResponder=_this$props5.panResponder,props=(0,_objectWithoutProperties2.default)(_this$props5,["delayLongPress","delayPressIn","delayPressOut","disabled","hitSlop","pressRetentionOffset","children","rippleContainerBorderRadius","testID","nativeID","accessible","accessibilityLabel","panResponder"]);var touchableProps={delayLongPress:delayLongPress,delayPressIn:delayPressIn,delayPressOut:delayPressOut,disabled:disabled,hitSlop:hitSlop,pressRetentionOffset:pressRetentionOffset,onPress:onPress,onPressIn:onPressIn,testID:testID,nativeID:nativeID,accessible:accessible,accessibilityLabel:accessibilityLabel,onPressOut:onPressOut,onLongPress:props.onLongPress?<span class="branch-0 cbranch-no" title="branch not covered" >onLongPress:</span>undefined,onLayout:onLayout};var containerStyle={borderRadius:rippleContainerBorderRadius};return _react.default.createElement(_TouchableWithoutFeedback.default,(0,_extends2.default)({},touchableProps,panResponder,{testID:testID,__self:this,__source:{fileName:_jsxFileName,lineNumber:293,columnNumber:7}}),_react.default.createElement(_Animated.default.View,(0,_extends2.default)({useNativeDriver:false},props,{__self:this,__source:{fileName:_jsxFileName,lineNumber:294,columnNumber:9}}),children,_react.default.createElement(_View.default,{style:[_Ripple.default.container,containerStyle],pointerEvents:"none",__self:this,__source:{fileName:_jsxFileName,lineNumber:296,columnNumber:11}},ripples.map(this.renderRipple))));}}]);return Ripple;}(_react.PureComponent);exports.default=Ripple;Ripple.defaultProps=(0,_extends2.default)({},_TouchableWithoutFeedback.default.defaultProps,{rippleColor:'rgba(0, 0, 0, .87)',rippleOpacity:0.3,rippleDuration:400,rippleSize:0,rippleContainerBorderRadius:0,rippleCentered:false,rippleSequential:false,rippleFades:true,disabled:false,displayUntilPressOut:true,onRippleAnimation:function <span class="fstat-no" title="function not covered" >onRippleAnimation(</span>animation,callback){<span class="cstat-no" title="statement not covered" >return animation.start(callback);}</span>,testID:'mb-ripple'});Ripple.propTypes=(0,_extends2.default)({},_Animated.default.View.propTypes,_TouchableWithoutFeedback.default.propTypes,{rippleColor:_propTypes.default.string,rippleOpacity:_propTypes.default.number,rippleDuration:_propTypes.default.number,rippleSize:_propTypes.default.number,rippleContainerBorderRadius:_propTypes.default.number,rippleCentered:_propTypes.default.bool,rippleSequential:_propTypes.default.bool,rippleFades:_propTypes.default.bool,disabled:_propTypes.default.bool,displayUntilPressOut:_propTypes.default.bool,onRippleAnimation:_propTypes.default.func,testID:_propTypes.default.string});</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sun Jul 11 2021 09:12:08 GMT-0600 (Mountain Daylight Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    