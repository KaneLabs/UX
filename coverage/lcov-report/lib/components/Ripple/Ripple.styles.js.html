
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for lib/components/Ripple/Ripple.styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">lib/components/Ripple</a> Ripple.styles.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>10/10</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">2x</span></td><td class="text"><pre class="prettyprint lang-js">var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=exports.radius=void 0;var _extends2=_interopRequireDefault(require("@babel/runtime/helpers/extends"));var _StyleSheet=_interopRequireDefault(require("react-native-web/dist/exports/StyleSheet"));var radius=10;exports.radius=radius;var styles={container:(0,_extends2.default)({},_StyleSheet.default.absoluteFillObject,{backgroundColor:'transparent',overflow:'hidden',zIndex:10}),ripple:{width:radius*2,height:radius*2,borderRadius:radius,overflow:'hidden',position:'absolute',zIndex:10}};var _default=styles;exports.default=_default;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sun Jul 11 2021 09:12:08 GMT-0600 (Mountain Daylight Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    