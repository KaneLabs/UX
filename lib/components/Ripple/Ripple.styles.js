var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=exports.radius=void 0;var _extends2=_interopRequireDefault(require("@babel/runtime/helpers/extends"));var _StyleSheet=_interopRequireDefault(require("react-native-web/dist/exports/StyleSheet"));var radius=10;exports.radius=radius;var styles={container:(0,_extends2.default)({},_StyleSheet.default.absoluteFillObject,{backgroundColor:'transparent',overflow:'hidden',zIndex:10}),ripple:{width:radius*2,height:radius*2,borderRadius:radius,overflow:'hidden',position:'absolute',zIndex:10}};var _default=styles;exports.default=_default;