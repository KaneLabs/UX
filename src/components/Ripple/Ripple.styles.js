import { StyleSheet } from 'react-native';

export const radius = 10;

const styles = {
  container: {
    ...StyleSheet.absoluteFillObject,

    backgroundColor: 'transparent',
    overflow: 'hidden',
    zIndex: 10,
  },

  ripple: {
    width: radius * 2,
    height: radius * 2,
    borderRadius: radius,
    overflow: 'hidden',
    position: 'absolute',
    zIndex: 10,
  },
};

export default styles;
