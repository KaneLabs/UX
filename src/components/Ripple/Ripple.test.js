import React from 'react';
import renderer from 'react-test-renderer';
import Ripple from './Ripple';

test('Ripple Renders', () => {
  const tree = renderer.create(<Ripple />).toJSON();
  expect(tree).toMatchSnapshot();
});
