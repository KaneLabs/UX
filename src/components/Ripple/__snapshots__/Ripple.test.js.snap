// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripple Renders 1`] = `
<View
  accessible={true}
  displayUntilPressOut={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onRippleAnimation={[Function]}
  onStartShouldSetResponder={[Function]}
  rippleCentered={false}
  rippleColor="rgba(0, 0, 0, .87)"
  rippleDuration={400}
  rippleFades={true}
  rippleOpacity={0.3}
  rippleSequential={false}
  rippleSize={0}
  testID="mb-ripple"
  useNativeDriver={false}
>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 10,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  />
</View>
`;
